### ISSUE TITLE
Changing view crashes

### ISSUE THREAD

**Issue by rappen** [Innofactor/PluginRegistration#60]
Trying to change view to By Entity or By Message and I get the following exception:
```
Unhandled Exception: System.SystemException

The given key was not present in the dictionary.

   at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Xrm.Sdk.PluginRegistration.MainControl.CreateCrmTreeNodes(CrmViewType view, Guid messageId, Guid messageEntityId, Boolean addToTree)
   at Xrm.Sdk.PluginRegistration.MainControl.LoadNodes(CrmViewType view)
```

**Comment by imranakram** [comment]
I've tried to reproduce it on 8.1, 8.2 and online environments but still it seems to work fine. You know how many plugins we have :) 

Others have been reporting this behavior as well. Do you have a chance to debug this or can I have access to an environment where you see this behavior?

**Comment by daryllabar** [comment]
Maybe as a stop gap, you wrap the CreateCrmTreeNodes in a try catch, where you spit out the key that was not found?  I'm wondering if it's something 9.+ related?  I've gotten it multiple times, but not always consistently.

**Comment by arow44** [comment]
Same issue on more environments. Will you help any log or further tests?

**Comment by CodyHarrison** [comment]
I've the same issue with CRM online 9.1.0000.11437 with UI. It worked before ...

**Comment by kendalls** [comment]
The issue has come and gone a bit in our v9.1 environments. It's back though so I tried debugging the plugin. We have preview features enabled and it turns out that the Microsoft.Dynamics.Forecasting.Plugins assembly has a couple of steps registered on invalid messages. The stack track looks like:
at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Xrm.Sdk.PluginRegistration.Wrappers.CrmEntityDictionary`1.get_Item(Guid id) in C:\Temp\PluginRegistration-master\PluginRegistration-master\Xrm.Sdk.PluginRegistration\Wrappers\CrmEntityDictionary.cs:line 81
   at Xrm.Sdk.PluginRegistration.MainControl.CreateCrmTreeNodes(CrmViewType view, Guid messageId, Guid messageEntityId, Boolean addToTree) in C:\Temp\PluginRegistration-master\PluginRegistration-master\Xrm.Sdk.PluginRegistration\MainControl.cs:line 769
   at Xrm.Sdk.PluginRegistration.MainControl.LoadNodes(CrmViewType view) in C:\Temp\PluginRegistration-master\PluginRegistration-master\Xrm.Sdk.PluginRegistration\MainControl.cs:line 1181

The plugin tries to create nodes for messageids and messageentityids that aren't in the m_org.Messages and m_org.MessageEntities dictionaries. For the sake of expediency, I just skipped processing the step if its messageid or messageentityid weren't in the corresponding dictionaries.
